Mathlib/CategoryTheory/Sites/EffectiveEpimorphic.lean

```diff
@@ -717,4 +717,141 @@ lemma effectiveEpi_iff_epi {X Y : C} (f : X ⟶ Y) : EffectiveEpi f ↔ Epi f :=
 
 end Epi
 
+noncomputable section Equivalence
+
+variable {D : Type*} [Category D] (e : C ≌ D) {B : C}
+
+variable {α : Type*} (X : α → C) (π : (a : α) → (X a ⟶ B)) [EffectiveEpiFamily X π]
+
+theorem effectiveEpiFamilyStructOfEquivalence_aux {W : D} (ε : (a : α) → e.functor.obj (X a) ⟶ W)
+    (h : ∀ {Z : D} (a₁ a₂ : α) (g₁ : Z ⟶ e.functor.obj (X a₁)) (g₂ : Z ⟶ e.functor.obj (X a₂)),
+      g₁ ≫ e.functor.map (π a₁) = g₂ ≫ e.functor.map (π a₂) → g₁ ≫ ε a₁ = g₂ ≫ ε a₂)
+    {Z : C} (a₁ a₂ : α) (g₁ : Z ⟶ X a₁) (g₂ : Z ⟶ X a₂) (hg : g₁ ≫ π a₁ = g₂ ≫ π a₂) :
+    g₁ ≫ (fun a ↦ e.unit.app (X a) ≫ e.inverse.map (ε a)) a₁ =
+    g₂ ≫ (fun a ↦ e.unit.app (X a) ≫ e.inverse.map (ε a)) a₂ := by
+  have := h a₁ a₂ (e.functor.map g₁) (e.functor.map g₂)
+  simp only [← Functor.map_comp, hg] at this
+  simpa using congrArg e.inverse.map (this (by trivial))
+
+/-- Equivalences preserve effective epimorphic families -/
+def effectiveEpiFamilyStructOfEquivalence : EffectiveEpiFamilyStruct (fun a ↦ e.functor.obj (X a))
+    (fun a ↦ e.functor.map (π a)) where
+  desc ε h := (e.toAdjunction.homEquiv _ _).symm
+      (EffectiveEpiFamily.desc X π (fun a ↦ e.unit.app _ ≫ e.inverse.map (ε a))
+      (effectiveEpiFamilyStructOfEquivalence_aux e X π ε h))
+  fac ε h a := by
+    simp only [Functor.comp_obj, Adjunction.homEquiv_counit, Functor.id_obj,
+      Equivalence.toAdjunction_counit]
+    have := congrArg ((fun f ↦ f ≫ e.counit.app _) ∘ e.functor.map)
+      (EffectiveEpiFamily.fac X π (fun a ↦ e.unit.app _ ≫ e.inverse.map (ε a))
+      (effectiveEpiFamilyStructOfEquivalence_aux e X π ε h) a)
+    simp only [Functor.id_obj, Functor.comp_obj, Function.comp_apply, Functor.map_comp,
+        Category.assoc, Equivalence.fun_inv_map, Iso.inv_hom_id_app, Category.comp_id] at this
+    rw [this]
+    exact e.toAdjunction.left_triangle_components_assoc _ _
+  uniq ε h m hm := by
+    simp only [Functor.comp_obj, Adjunction.homEquiv_counit, Functor.id_obj,
+      Equivalence.toAdjunction_counit]
+    have := EffectiveEpiFamily.uniq X π (fun a ↦ e.unit.app _ ≫ e.inverse.map (ε a))
+      (effectiveEpiFamilyStructOfEquivalence_aux e X π ε h)
+    specialize this (e.unit.app _ ≫ e.inverse.map m) fun a ↦ ?_
+    · rw [← congrArg e.inverse.map (hm a)]
+      simp
+    · simp only [← this, Functor.comp_obj, Functor.map_comp, Equivalence.fun_inv_map,
+        Functor.id_obj, Category.assoc, Iso.inv_hom_id_app, Category.comp_id]
+      erw [e.toAdjunction.left_triangle_components_assoc]
```

Comment:
```suggestion
      rw [reassoc_of% e.functor_unit_comp]
```
There are a lot of reassoc attributes missing in `Equivalence.lean`, so if you like you can go ahead and add them, otherwise I will get around to it eventually.

Reply:
I can add the them. Should every lemma which is an equality of two morphisms have the `reassoc` attribute, and `reassoc (attr := simp)` if it's a simp lemma?

Reply:
Thanks! I think it's fine to do it as part of this PR.